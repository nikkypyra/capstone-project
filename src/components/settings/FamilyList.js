import React, { useEffect, useState } from 'react'
import styled from 'styled-components/macro'
import { db } from '../../firebase'
import PropTypes from 'prop-types'
import DeleteButton from '../buttons/DeleteButton'

FamilyList.propTypes = {
  user: PropTypes.object.isRequired,
}

export default function FamilyList({ user }) {
  const [allUsers, setAllUsers] = useState([])
  const chosenUser =
    allUsers.find((individual) => individual.id === user.id) || {}
  const familyMembers = chosenUser.family || []
  useEffect(() => {
    db.collection('users').onSnapshot((snapshot) => {
      const allUsers = snapshot.docs.map((doc) => ({
        id: doc.id,
        ...doc.data(),
      }))
      setAllUsers(allUsers)
    })
  }, [])

  return (
    <>
      {familyMembers.map((person, index) => (
        <FamilyWrapper key={index}>
          <div className="person">
            <p>{person}</p>
          </div>
          <div className="delete">
            <DeleteButton />
          </div>
        </FamilyWrapper>
      ))}
    </>
  )
}

const FamilyWrapper = styled.section`
  margin: 16px 0;
  display: flex;
`
