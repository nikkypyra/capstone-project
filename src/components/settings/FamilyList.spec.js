import React from 'react'
import { render, cleanup } from '@testing-library/react'
import FamilyList from './FamilyList'
import { MemoryRouter as Router } from 'react-router-dom'

afterEach(cleanup)

const allUsersMock = [
  {
    id: '1',
    email: 'user@user.com',
    family: ['mom@pawlog.com', 'dad@pawlog.com'],
  },
]
const userMock = {
  id: '1',
  email: 'user@user.com',
  family: ['mom@pawlog.com', 'dad@pawlog.com'],
}

const deleteFamily = jest.fn()

describe('<FamilyList />', () => {
  it('should render correctly', () => {
    const { asFragment } = render(
      <Router>
        <FamilyList
          user={userMock}
          allUsers={allUsersMock}
          deleteFamily={deleteFamily}
        />
      </Router>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
