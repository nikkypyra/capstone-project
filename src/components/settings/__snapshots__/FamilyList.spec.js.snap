// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FamilyList /> should render correctly 1`] = `
<DocumentFragment>
  <section
    class="FamilyList__Animation-sc-1lx9e3m-4 iEWBLl"
  >
    <h2
      class="FamilyList__Title-sc-1lx9e3m-0 eggmlq"
    >
      Family Members
    </h2>
    <section
      class="FamilyList__FamilyWrapper-sc-1lx9e3m-1 lZgvI"
    >
      <div
        class="FamilyList__Marker-sc-1lx9e3m-2 dEWogB"
      >
        <img
          alt=""
          src="user.png"
        />
      </div>
      <div
        class="person"
      >
        <h4>
          mom@pawlog.com
        </h4>
      </div>
      <div
        class="delete"
        data-cy="delete-family"
      >
        <div
          class="sc-AxjAm ljvhLh"
        >
          <img
            alt="delete"
            src="delete.png"
          />
        </div>
      </div>
    </section>
    <section
      class="FamilyList__FamilyWrapper-sc-1lx9e3m-1 lZgvI"
    >
      <div
        class="FamilyList__Marker-sc-1lx9e3m-2 dEWogB"
      >
        <img
          alt=""
          src="user.png"
        />
      </div>
      <div
        class="person"
      >
        <h4>
          dad@pawlog.com
        </h4>
      </div>
      <div
        class="delete"
        data-cy="delete-family"
      >
        <div
          class="sc-AxjAm ljvhLh"
        >
          <img
            alt="delete"
            src="delete.png"
          />
        </div>
      </div>
    </section>
  </section>
</DocumentFragment>
`;
